### ISSUE TITLE
[Custom Calculators] The required option "currency" is missing.

### ISSUE THREAD

**Issue by rmNyro** [Sylius/Sylius#8140]
Hi (again :p),

Context first :
I reproduced the error on a very fresh (I only added a route for another issue that has absolutely nothing to do with this and doesn't generate any error) install of sylius beta-2 (from the download instructions using composer). I tried to implement a custom calculator and so I pretty much copied/pasted the `flat_rate` calculator and its configuration type (because the doc is broken and hard to work with).

Here comes the code :
```yml
# src/AppBundle/Resources/config/services.yml
services:
    app.shipping_calculator.by_weight:
        class: AppBundle\Shipping\MyCustomCalculator
        tags:
            - { name: sylius.shipping_calculator, form-type: AppBundle\Shipping\Form\Type\MyCustomCalculatorConfigurationType,
                calculator: "my_calculator", label: "My Custom Calculator" }
```
```php
// src/AppBundle/Shipping/Form/Type/ShippingByWeightConfigurationType.php
namespace AppBundle\Shipping\Form\Type;

use Sylius\Bundle\MoneyBundle\Form\Type\MoneyType;
use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\FormBuilderInterface;
use Symfony\Component\OptionsResolver\OptionsResolver;
use Symfony\Component\Validator\Constraints\NotBlank;
use Symfony\Component\Validator\Constraints\Type;

class MyCustomCalculatorConfigurationType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder
            ->add('amount', MoneyType::class, [
                'label' => 'my amount test',
                'constraints' => [
                    new NotBlank(['groups' => ['sylius']]),
                    new Type(['type' => 'integer', 'groups' => ['sylius']]),
                ],
                'currency' => $options['currency'],
            ])
        ;
    }

    public function configureOptions(OptionsResolver $resolver)
    {
        $resolver
            ->setDefaults([
                'data_class' => null,
            ])
            ->setRequired(['currency'])
            ->setAllowedTypes('currency', 'string')
        ;
    }

    public function getName()
    {
        return 'app_shipping_my_calculator';
    }
}
```
```php
// src/AppBundle/Shipping/ShippingByWeightCalculator.php
namespace AppBundle\Shipping;

use Sylius\Component\Shipping\Calculator\CalculatorInterface;
use Sylius\Component\Shipping\Model\ShipmentInterface;

final class MyCustomCalculator implements CalculatorInterface
{
    public function calculate(ShipmentInterface $subject, array $configuration)
    {
        return (int) $subject->getShippingWeight() + $configuration['amount'];
    }

    public function getType()
    {
        return 'my_calculator';
    }
}
```

So either I didn't understand what `currency` is supposed to be, or I miss one step, or there's a bug.
If I remove `currency`, everything works just fine.

Keep up the great work and please refactor the frontend default layout logic, it's quite hard to follow sometimes. Simplicity is good especially if it has a learning value (as I suppose it has). ^^
If I have the time I'll make a PR with a webpack version (which I currently use in my project).

Cheers.

**Comment by lchrusciel** [comment]
The currency is supposed to be a currency code, which is needed to properly display values but it is not stored anyway. Sort of a UI feature only. Check `ChannelBasedFlatRateConfigurationType` for example usage of type. 

Closing.
